Service/ContentTypeService.php

```diff
@@ -409,4 +416,60 @@ public function getAllTypes()
         $this->loadEnvironment();
         return implode(',', array_keys($this->contentTypeArrayByName));
     }
+    
+    /**
+     * Export a content type in Json format
+     *
+     */
+    public function createContentTypeFromJson(File $jsonFile, Environment $environment)
+    {
+        $em = $this->doctrine->getManager();
+        /** @var ContentTypeRepository $contentTypeRepository */
+        $contentTypeRepository = $em->getRepository('EMSCoreBundle:ContentType');
+        $fileContent = file_get_contents($jsonFile->getRealPath());
+        
+        $encoders = array(new JsonEncoder());
+        $normalizers = array(new JsonNormalizer());
+        $serializer = new Serializer($normalizers, $encoders);
+        /**@var ContentType $contentType */
+        $contentType = $serializer->deserialize(
+            $fileContent,
+            "EMS\CoreBundle\Entity\ContentType",
+            'json'
+        );
+        $contentType->setEnvironment($environment);
+        $contentType->setActive(false);
+        $contentType->setDirty(true);
+        $contentType->getFieldType()->updateAncestorReferences($contentType, null);
+        $contentType->setOrderKey($contentTypeRepository->maxOrderKey() + 1);
+        
+        $this->persist($contentType);
+        return $contentType->getId();
+    }
+    
+    /**
+     * Export a content type in Json format
+     */
+    public function exportToJson(ContentType $contentType)
+    {
+        //Sanitize the CT
+        $contentType->setCreated(null);
+        $contentType->setModified(null);
+        $contentType->getFieldType()->removeCircularReference();
+        $contentType->setEnvironment(null);
+        
+        //Serialize the CT
+        $encoders = array(new JsonEncoder());
+        $normalizers = array(new JsonNormalizer());
+        $serializer = new Serializer($normalizers, $encoders);
+        $jsonContent = $serializer->serialize($contentType, 'json');
+        $response = new Response($jsonContent);
```

Comment:
Respones are only usefull for Controllers that render a web page (Response).
This function should return the $jsonContent, and the controller using this function can then generate the response.